docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -1088,3 +1092,210 @@ function sitenow_update_8032() {
+  $query->fields('n', ['entity_id', 'delta'])
+    ->condition('layout_builder__layout_section', '%uiowa_hero%', 'LIKE');
+  $results = $query->execute();
+
+  // Create some controllers for later.
+  $block_storage = \Drupal::service('entity_type.manager')
+    ->getStorage('block_content');
+  $node_storage = \Drupal::service('entity_type.manager')
+    ->getStorage('node');
+  $database = \Drupal::database();
+  $use_controller = new InlineBlockUsage($database);
+
+  // UUIDs of two homepage hero components not to be updated.
+  $no_updates = [
+    'fe6bc0f1-7fcd-4802-922e-08c60eff58c6',
+    'a3f324a0-53ea-4336-9f68-224f358a74fd',
+  ];
+  // @todo Batch the work.
+  foreach ($results as $record) {
+    $nid = $record->entity_id;
+    $section_delta = $record->delta;
+
+    // Grab the referenced node, its layout, and the section.
+    $node = $node_storage->load($nid);
+    $layout = $node->get('layout_builder__layout');
+    $section = $layout->get($section_delta)->getValue()['section'];
+
+    // Should be only one, but iterate in case there are more.
+    // Also easier to grab this way without knowing the comp uuid.
+    $components = $section->getComponents();
+    foreach ($components as $component) {
+      $component_arr = $component->toArray();
+      // Check that it's a hero, and not in our no-update list.
+      if (!in_array($component_arr['uuid'], $no_updates) &&
+        $component_arr['configuration']['id'] === 'inline_block:uiowa_hero') {
+
+        // Load the currently associated hero block revision.
+        $rev_id = $component_arr['configuration']['block_revision_id'];
+        $hero_block = $block_storage->loadRevision($rev_id);
+
+        // Continue only if there is no video associated w/the block.
+        if ($hero_block->get('field_uiowa_hero_video')->isEmpty()) {
+          // Remove the old component by its uuid.
+          $section->removeComponent($component_arr['uuid']);
+
+          // Grab field values relavent to the new banner.
+          $excerpt = $hero_block->get('field_uiowa_hero_excerpt')->getValue()[0];
+          $image = $hero_block->get('field_uiowa_hero_image')->getValue()[0];
+          $title = $hero_block->get('field_uiowa_hero_title')->getValue()[0];
+          $link = NULL;
+
+          // Grab and process component styles.
+          $comp_styles = $component_arr['additional']['layout_builder_styles_style'];
+          // Set the desired default styles, and override if necessary.
+          $new_styles = [
+            'image',
+            'banner_centered',
+            'banner_medium',
+            'headline_bold_serif',
+          ];
+          // Can add to this if more style transfers are needed.
+          foreach ($comp_styles as $style) {
+            switch ($style) {
+              // Replace default centered with left aligned.
+              case 'block_style_hero_left_aligned':
+                $new_styles[1] = 'banner_centered_left';
+                break;
```

Comment:
This code sets the style options for banners created out of legacy hero blocks.

I think `banner_centered` is accurate if no style is set (like the banner on coronavirus.uiowa.edu), but the hero's `block_style_hero_left_aligned` should maybe be updated to `banner_lower_left` to better fit existing styles (like americanstudies.prod.drupal.uiowa.edu).

Reply:
@cory-skeers I see you just updated it lower left, but I would prefer the centered as the default update.